### PR TITLE
Make sure that parse_xyz returns a dict xyz

### PR DESCRIPTION
**Pull request by kfir4444**

### CODE REVIEW

File: arc/parser.py [hunk 1]
```diff
@@ -231,10 +231,10 @@ def parse_geometry(path: str) -> Optional[Dict[str, tuple]]:
         raise InputError(f'Could not find file {path}')
     if path.endswith('.yml'):
         content = read_yaml_file(path)
-        if isinstance(content, dict) and 'xyz' in content.keys():
-            return content['xyz']
-        if isinstance(content, dict) and 'opt_xyz' in content.keys():
-            return content['opt_xyz']
+        if 'xyz' in content.keys():
```
Comment (alongd) [inline arc/parser.py:234]: why removing the safety check `if isinstance(content, dict)`? If we're reading a YAML what doesn't hold a dict, this like will crash

File: arc/parser.py [hunk 2]
```diff
@@ -231,10 +231,10 @@ def parse_geometry(path: str) -> Optional[Dict[str, tuple]]:
...
+            return content['xyz'] if isinstance(content['xyz'], dict) else str_to_xyz(content['xyz'])
```
Comment (alongd) [inline arc/parser.py:235]: 👍

File: arc/parser.py [hunk 3]
```diff
@@ -231,10 +231,10 @@ def parse_geometry(path: str) -> Optional[Dict[str, tuple]]:
         raise InputError(f'Could not find file {path}')
     if path.endswith('.yml'):
         content = read_yaml_file(path)
-        if isinstance(content, dict) and 'xyz' in content.keys():
```
Comment (alongd) [inline arc/parser.py:234]: Can you please modify the commit message to "Make **sure**...."

**Review by alongd — COMMENTED** [review]
Thanks! Added minor comments

File: arc/parser.py (hunk 3 above)
Comment (alongd) [reply arc/parser.py:234]: I think this wasn't addressed

File: arc/parser.py (hunk 1 above)
Comment (alongd) [reply arc/parser.py:234]: I think this wasn't addressed

File: arc/parser.py (hunk 3 above)
Comment (Lilachn91) [inline arc/parser.py:234]: Why did you remove isinstance statement here, and not keep it like below?

**Review by alongd — APPROVED** [review]

**Review by Lilachn91 — COMMENTED** [review]
Good job :)
